### PR TITLE
Remote Debugging Filter Change and Azure Scenario Support

### LINKED ISSUE #673: [Local Debugging] VS can actually attach to PowerShell ISE
In Attaching window, PowerShell ISE is excluded as a attachable type for debugging.

### PR DESCRIPTION
**Pull request by MrayerMSFT**
Changed attachable logic to use Get-PSHostProcessInfo as the filter for finding attachable processes. This is much more accurate than looking for the powershell.exe module. Added an alternative remote debugging transport which utilizes SSL. This will allow users to connect to Azure VMs. Also added the ability to specify a port. Not a very likely scenario, but Azure lets you change the PowerShell connection port so it may be useful to some users.

Resolves #673 

@EricMSFT @AndreSayreMSFT @HuanhuanSunMSFT

### CODE REVIEW

**Comment by ericdevg** [comment]
Good work!

File: PowerShellTools.HostService/ServiceManagement/Debugging/PowerShellDebuggingService.cs [hunk 1]
```diff
@@ -181,30 +182,36 @@ public bool IsAttachable(uint pid)
...
-                    IsWow64Process(process.Handle, out is64Bit);
-
-                    // cannot examine a 64 bit process' modules from a 32 bit process
-                    if (!Environment.Is64BitProcess && is64Bit)
-                    {
-                        return false;
-                    }
-                    else if (process != null)
+                    // see if the process is a PS host
+                    if (InvokeScript(_currentPowerShell, string.Format("Get-PSHostProcessInfo -Id {0}", pid)).Count > 0)
```
Comment (ASayre) [inline PowerShellTools.HostService/ServiceManagement/Debugging/PowerShellDebuggingService.cs]: Can we use .Any() ?

File: PowerShellTools/DebugEngine/Remote/IPowerShellToolsPortSupplier.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+namespace PowerShellTools.DebugEngine.Remote
+{
+    /// <summary>
+    /// Interface for all remote debugging port suppliers that this tool implements. Enforces the
+    /// need for said port suppliers to specify whether or not the -UseSSL parameter should be used
+    /// when entering into a remote session with a machine.
+    /// </summary>
+    interface IPowerShellToolsPortSupplier : IDebugPortSupplier2, IDebugPortSupplierDescription2
+    {
+        bool UsesSSL();
```
Comment (ASayre) [inline PowerShellTools/DebugEngine/Remote/IPowerShellToolsPortSupplier.cs:18]: Q, and potentially stylistic,: Method vs Property?

Comment (MrayerMSFT) [inline PowerShellTools/DebugEngine/Remote/IPowerShellToolsPortSupplier.cs:18]: You can't have private properties in an interface, and whether or not a port supplier uses SSL should not be modifiable by someone else. (And yes, I realize I could do something like [this](http://www.functionx.com/csharp/examples/interface1.htm) but I prefer this way) It also allows for flexibility whenever a potential future supplier needs to define whether or not they use SSL. Maybe for them it won't be a true/false situation, it might be based one some conditions, etc. Lastly, I think using a method is more true to the purpose of an interface which is to provide a contract that other objects can use to communicate with your object.

TL;DR: pure personal choice

**Comment by ASayre** [comment]
looks good
